argocd/scripts/sync_and_wait.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import json
+import time
+import argparse
+import os
+#
+# python3 wait_for_sync.py --wait-for=30 --application=default --target=3f9c8c3c02d8573901bd4c4e0c9ddfe7d7dcddf5 --argocd-url https://argocd.metering-shared-non-prod.ovotech.org.uk/
+#
+def sync_request(endpoint, token, application):
+
+    try:
+        headers = { 'Authorization' : 'Bearer %s' % token }
+        res = requests.post(endpoint.rstrip("/") + f'/api/v1/applications/{application}/sync', headers=headers, timeout = 10, verify=True)
+        if res.status_code != 200:
+            print(f"ERROR: Non-200 response ({res.status_code}): {res.json()}.")
+            return False
+        
+        data = res.json()
+        cluster_sync_status = data['status']['sync']['status']
+
+        if os.environ.get('ARGOCD_ORB_DEBUG'):
+            print_debug(data)
+
+        if cluster_sync_status == "Synced":
+            print(f'Sync request successful, application was in sync')
+            return True
+        else:
+            print(f'Sync request successful, application was not in sync')
+            return False
+
+    except ValueError as e:
+        print(f'ERROR: Decoding JSON has failed: {e}')
+        return False
+        
+    except Exception as e:
+        print(f'ERROR: Request failed: {e}')
+        return False
+
+def is_cluster_insync(endpoint, token, application, target_revision):
```

Comment:
I think we've duplicated some python code. Could we combine in into one python file, and then just invoke either wait or sync based on command line arguments?

Reply:
Good point, given this ended up being a minimal change in the end, I've now put the sync into the same job as before. No new job or scripts